pensieve/tests/test_config.py

```diff
@@ -136,3 +155,47 @@ class MyCoolClass:
             )
 
         assert "Could not locate MyCoolClass spam" in str(e.value)
+
+    def test_unknown_statistic_failure(self, experiments):
+        config_str = dedent(
+            """
+            [metrics]
+            weekly = [{metric = "spam", treatment = "unknown_stat"}]
+
+            [metrics.spam]
+            data_source = "main"
+            select_expression = "1"
+            """
+        )
+
+        with pytest.raises(ValueError) as e:
+            spec = config.AnalysisSpec.from_dict(toml.loads(config_str))
+            spec.resolve(experiments[0])
+
+        assert "Statistic unknown_stat does not exist" in str(e)
+
+    def test_overwrite_statistic(self, experiments):
+        config_str = dedent(
+            """
+            [metrics]
+            weekly = [{metric = "spam", treatment = "bootstrap_mean"}]
+
+            [metrics.spam]
+            data_source = "main"
+            select_expression = "1"
+
+            [statistics.bootstrap_mean]
+            num_samples = 10
+            branches = ["a", "b"]
```

Comment:
This is how custom parameters can be set for existing statistics. Adding new statistics will require a PR.

Reply:
I think the parameters will often be specific to the context of the metric -- e.g. breakpoints for thresholds, so we'll need to able to represent that.

I think "branches" probably isn't something we should expose.

Reply:
I see, so would we want the config for a metric look something like this?

```toml
[metrics.test_metric]
...
[metrics.test_metric.statistics.tests_statistic]
...
```

or 

```toml
[metrics.test_metric]
...
treatment = {
   name = "test_statistic"
   param1 = 123
   ...
}
```